### PR TITLE
refactor: triage untestable DCOS code

### PR DESCRIPTION
**Pull request by jackfrancis**
**Reason for Change**:

Move untestable DC/OS code to a distinct go file marked as "skip unit tests here". Also includes a refactor of the `generateIPList` func (to `generateConsecutiveIPsList`) to include basic edge case error checking.

**Issue Fixed**:

**Requirements**:

- [ ] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Notes**:

### CODE REVIEW

File: pkg/engine/engine.go [hunk 1]
```diff
@@ -119,16 +118,25 @@ func validateDistro(cs *api.ContainerService) bool {
 	return true
 }
 
-func generateIPList(count int, firstAddr string) []string {
+// generateConsecutiveIPsList takes a starting IP address and returns a string slice of length "count" of subsequent, consecutive IP addresses
+func generateConsecutiveIPsList(count int, firstAddr string) ([]string, error) {
```
Comment (jackfrancis) [inline pkg/engine/engine.go]: I actually made the effort to refactor + test this function because it was appalling in its current state

File: pkg/engine/engine_dcos.go [hunk 2]
```diff
@@ -0,0 +1,250 @@
+//+build !test
```
Comment (jackfrancis) [inline pkg/engine/engine_dcos.go]: @mboersma I was hoping that this pragma would remove this file from unit test coverage calculation, which should have moved the needle on our % coverage by a wee bit. Do you know of any better ideas?

Comment (mboersma) [reply pkg/engine/engine_dcos.go]: You could add it to the `ignore:` section in .codecov/codecov.yml.

Comment (CecileRobertMichon) [reply pkg/engine/engine_dcos.go]: does `//+build !test` do anything?

File: pkg/engine/template_generator.go [hunk 3]
```diff
@@ -10,7 +10,6 @@ import (
 	"encoding/json"
 	"fmt"
 	"runtime/debug"
-	"strconv"
```
Comment (CecileRobertMichon) [inline pkg/engine/template_generator.go]: did you run `make generate` on these changes?

Comment (jackfrancis) [reply pkg/engine/template_generator.go]: Yes, does something look off?

File: pkg/engine/engine_dcos.go (hunk 2 above)
Comment (jackfrancis) [reply pkg/engine/engine_dcos.go]: It tells the compiler not to include this when running unit tests

Comment (CecileRobertMichon) [reply pkg/engine/engine_dcos.go]: but does it actually work? I understood your comment above to mean that it doesn't

Comment (jackfrancis) [reply pkg/engine/engine_dcos.go]: It didn't seem to influence our current test coverage toolchain, but I'm not sure that our current test coverage calculation implementation is vanilla go-idiomatic. So basically: I left this in place as a way of formally declaring intent for these funcs in a go-idiomatic fashion. So at minimum it's basically a code comment at this point.

File: pkg/engine/template_generator.go (hunk 3 above)
Comment (CecileRobertMichon) [reply pkg/engine/template_generator.go]: I don't see a templates_generated.go diff

Comment (jackfrancis) [reply pkg/engine/template_generator.go]: What part of this changeset do we expect to see reflected in `templates_generated.go`?

Comment (CecileRobertMichon) [reply pkg/engine/template_generator.go]: I guess none, I thought I saw a parts/ change earlier but it's actually all pkg/

**Comment by CecileRobertMichon** [comment]
@jackfrancis is this ready for final review?

File: pkg/engine/engine.go [hunk 4]
```diff
@@ -119,16 +118,25 @@ func validateDistro(cs *api.ContainerService) bool {
...
 	ipaddr := net.ParseIP(firstAddr).To4()
 	if ipaddr == nil {
-		panic(fmt.Sprintf("IPAddr '%s' is an invalid IP address", firstAddr))
```
Comment (mboersma) [inline pkg/engine/engine.go]: Every time you remove a `panic()`, you gain karma. :-)
